addons/account/models/account_move.py

```diff
@@ -2107,11 +2108,11 @@ def dirty():
             if key not in existing_after
         }
         to_write = {
-            existing_after[key]: values
+            line: values
             for key, values in needed_after.items()
-            if key in existing_after
-            and any(
-                self.env['account.move.line']._fields[fname].convert_to_write(existing_after[key][fname], self)
+            for line in existing_after[key]
```

Comment:
It's very nice to remove the hack with `_('(Discount)')`, but this doesn't totally makes sense :thinking: 
One wouldn't want to update all the lines with the same key, otherwise the same balance can be written on multiple lines instead of one.

It only works because of this https://github.com/odoo/odoo/commit/932d0382b5a4a5d0d91dd532940bc2ed8d0c13da, making any change on the taxes go in `to_create`+`to_delete` because the `display_type` will always be different than `tax` (either `product` or `epd` or `rounding`)


In the end, I believe we can keep it for any changes that are being done for eventual non monetary fields.
It might raise an exception in some corner cases.

Reply:
Thanks for the review, I agree on the comment.
After almost 2 years of having this implementation (`_sync_dynamic_line`) in production it seems the only (problematic?) case of duplicate key is this one, I hope we won't have any surprises :sweat_smile: